robottelo/ui/base.py

```diff
@@ -64,31 +65,72 @@ def find_element(self, locator):
             )
         return None
 
-    def search_entity(self, element_name, element_locator, search_key=None,
-                      katello=None, button_timeout=15, result_timeout=15):
+    def _search_locator(self):
+        """Specify element name locator which should be used in search
+        procedure
+
+        """
+        raise NotImplementedError(
+            'Subclasses must return locator of element to search')
+
+    def _search_init(self, search_key=None, is_katello=None, button_timeout=15,
+                     result_timeout=15):
+        """Initialize every optional parameter for search procedure which can
+        be overridden for any unique case"""
+        if self.is_entity_delete:
+            button_timeout = 3
+            result_timeout = 1
+        return {
+            'search_key': search_key,
+            'is_katello': is_katello,
+            'button_timeout': button_timeout,
+            'result_timeout': result_timeout,
+        }
+
+    def navigate_to_entity(self):
+        """Perform navigation to main page for specific entity"""
+        raise NotImplementedError('Subclasses must implement navigator method')
+
+    def search(self, element_name):
         """Uses the search box to locate an element from a list of elements."""
-        search_key = search_key or 'name'
-
-        prefix = 'kt_' if katello else ''
+        # Navigate to the page
+        self.navigate_to_entity()
+
+        # Provide search criterions or use default ones
+        optional_parameters = self._search_init()
```

Comment:
Do we need this `_search_init` function? Do you think we could move that logic here (it's not much really)?